### ISSUE TITLE
[Change Request]: Add array of existing maps to state message

### ISSUE THREAD

**Issue by cpaulsynaos** [VDA5050/VDA5050#27]
We suggest to extend the AGV state message (5.6. Topic: State) by adding a an array **maps** and a JSON-object **map** that **maps** consists of.
This array should contain the list of maps available on the AGV consisting of mapId and information if a map is currently used. 

#### Usage of the field:
Some AGVs are already able to keep more than one map at once. 
Master control therefore needs to know
- what maps are present on the AGV
- which of the maps present is currently being active / used on the AGV
- if no map exists (yet) on the AGV

#### Structure:
Object Structure | Data Type | Description
------------------ | ----------- | -------------
header|N/A|Each JSON starts with a header. More information can be found in VDA 5050,  5.2 Protocol Header.
...|...|_[existing properties for state message; see VDA 5050, 5.6 Topic: State]_
**maps [map]**|array|Array of maps
map {|JSON-object|JSON object with the map information
mapId|uint32|ID of the map
active|boolean|**True**: Indicates this map is currently active / used on the AGV. **False**: Indicates this map is not currently active on the AGV and thus could be activated or deleted by request. At most one map can have this flag set to true
}||

#### Example:
```
{
…,
    "maps": [
        {
            "mapId": "1619becf-90f1-4152-b0f7-9b88893fb0ca",
            "active": true
        },
        {
            "mapId": "13467deb-f89a-5561-f96e-11bbfe3cad2b",
            "active": false
        }
    ]
}
```

**Comment by KIT-IFL** [comment]
Hello @cpaulsynaos,

the general idea was accepted and will be moved to the development version.

Just one small question: Should this be a mandatory field or an optional field?

**Comment by AntonDueck** [comment]
How does this fits to the issue #9 

is the active map changed by the AGV or defines the MC with an instant action  which map to activate ( the * activated or deleted by request* is hard to understand)

**Comment by cpaulsynaos** [comment]
Right, slightly misunderstandable as map changing is / will be part of upcoming discussions about "map distribution".
We would propose MC remotely ativating the map on an AGV via instant action (or dedicated map action topic)

**Comment by cpaulsynaos** [comment]
Related pull request: https://github.com/VDA5050/VDA5050/pull/66

**Comment by AntonDueck** [comment]
if the MC is activating the MapId and only one can be activ, then in the order is only one MapId otherwise the vehicle can not change the activ MapId in the state. so you add the mapID to order and remove it from the nodesPosition and your payload get smal. and the issue #9 has a total wrong think of mapId.

**Comment by ConstantinUp** [comment]
Maps are part of the released version 2.1. This issue should be resolved.
